### PR TITLE
Separate actor name and role

### PR DESCRIPTION
**Pull request by sbstnc**
Moved actor role to label2. Simplifies code and gives skinners greater
flexibility.

### CODE REVIEW

**Comment by HitcherUK** [comment]
@BigNoid Any decision as to whether this is I* or J* material?

Thanks.

**Comment by ronie** [comment]
@toolpunk could you please squash the two commits?

File: addons/skin.confluence/720p/DialogVideoInfo.xml [hunk 1]
```diff
@@ -892,6 +892,7 @@
 							<texture border="5">button-nofocus.png</texture>
 						</control>
 						<control type="label">
+							<description>Fname Sname as Role</description>
 							<left>10</left>
```
Comment (phil65) [inline addons/skin.confluence/720p/DialogVideoInfo.xml]: typos in description

Comment (toolpunk) [inline addons/skin.confluence/720p/DialogVideoInfo.xml]: I didn't want to type out first name and surname, which I now did. D'oh!

**Comment by BigNoid** [comment]
@HitcherUK I*** material as far as I'm concerned

**Comment by ronie** [comment]
@toolpunk please squash and rebase. :-)

**Comment by toolpunk** [comment]
@ronie Will do tomorrow. Not at home right now.

**Comment by toolpunk** [comment]
@ronie Is there a tutorial that's actually working? No matter what I do, I end up with conflicts, half complete rebases etc. This really sucks...

**Comment by ronie** [comment]
i'm sure there are a few GIT experts on the team, however i'm not one of them. sorry :-(

**Comment by HitcherUK** [comment]
@toolpunk Maybe ping Montellese for some help?

**Comment by Montellese** [comment]
@toolpunk: What exactly are you trying to do and what is the problem? If there weren't any conflicts github would be able to merge the PR directly so the chance that you run into at least a few conflicts during a rebase is basically a given. You need to resolve these conflicts manually by editing the conflicting files.

Another possibility is that you checkout current master, apply your changes to the current code (since you only have very few changes) in the same two commits and push that work to your `nfo-dlg-actor-role` remote git branch.

**Comment by toolpunk** [comment]
Well, this is pretty much the issue I'm dealing with. I start rebasing, resolve all conflicts and continue. Then the very same conflicts pop up again, I solve them again and continue. This goes on for a while and until I end up with a conflict in skin.re-touched which I didn't touch at all (no pun intended).

I suppose I could just merge master and re-apply the changes... but as far as I can tell, rebasing is preferable to merging since it's supposed to be invisible in the commit history. And even then I need to figure out a way to actually squash those commits (and the merge?).

It doesn't seem so simple once they're pushed. I understand I can work around this using git push -f (breaking currently non-existant forks), but I don't even get to that point. Some people suggest an interactive rebase while others prefer git reset --soft HEAD~X and I can't get either of these methods to do what I want. The interactive pick squash squash rebase takes me right back to rebase hell, and the reset approach doesn't seem to actually remove the previous commits.

**Comment by Montellese** [comment]
OK I pulled your branch and this is what I did to get it on top of current master:
1. `git checkout nfo-dlg-actor-role`
2. `git pull --rebase upstream master`
3. Fix the two conflicts in `xbmc/video/dialogs/GUIDialogVideoInfo.cpp`
4. `git add xbmc/video/dialogs/GUIDialogVideoInfo.cpp`
5. `git rebase --continue`
6. Fix the conflict in `xbmc/video/dialogs/GUIDialogVideoInfo.cpp`
7. `git add xbmc/video/dialogs/GUIDialogVideoInfo.cpp`
8. `git rebase --continue`

At this point your two commits are on top of current master. You can compile and test them.
If you want to squash the two commits or split the changes to confluence from the changes to CGUIDialogVideoInfo you can do
1. `git reset --soft HEAD~2`
2. `git reset HEAD`

Now you can either commit both files into a single commit or each file into a separate commit (which personally I'd prefer). Once that is done you can force push the changes to your remote branch:
1. `git push -f origin nfo-dlg-actor-role`

You can see my result at https://github.com/Montellese/xbmc/commits/nfo-dlg-actor-role.

**Comment by toolpunk** [comment]
Alright, I'll probably create a fresh clone of my own fork then, because this looks very much like what I've been trying. Seems like something else is causing all this trouble then. Thanks for the help!

**Comment by toolpunk** [comment]
Finally got around to do it. Seems like my text editor messed up the indentations again though. Fantastic. I <3 Windows and its tools.

**Comment by HitcherUK** [comment]
Thanks again for this - my skin is all ready to use them.

**Comment by HitcherUK** [comment]
@ronie Is this good to go yet?

**Comment by ronie** [comment]
as mentioned by @toolpunk the indenting is incorect.

**Comment by HitcherUK** [comment]
@toolpunk Any ETA? Thanks.

**Comment by toolpunk** [comment]
So sorry. Totally forgot about this. I applied the required fix a few moments ago. Should be all good now.

**Comment by BigNoid** [comment]
The typos are still in Confluence commit ;)

**Comment by toolpunk** [comment]
Those aren't typos, but commonly understood abbreviations.

**Comment by HitcherUK** [comment]
Many thanks for adding this.

**Comment by BigNoid** [comment]
Lol you do realize this is an international project and not just for native english speakers.
I just saw that @phil65 comment was not on an outdated diff.

*unsubscribes*

File: addons/skin.confluence/720p/DialogVideoInfo.xml (hunk 1 above)
Comment (MartijnKaijser) [inline addons/skin.confluence/720p/DialogVideoInfo.xml]: might as well type it out no one else will guess this

**Comment by toolpunk** [comment]
Changed it to `Actor and Role` now.

I really don't get it though. Why would anyone in their right mind try to pick a fight over an additional, optional comment meant to help people. I'm afraid those who didn't understand the last wording won't understand the new one either. Yet I still believe it's better to leave the comments in there as they may be of help to *some* people.

**Comment by ronie** [comment]
let's get this over with :-)

**Pull request merged**